stdlib/benchmarks/builtin/bench_hash.mojo

```diff
@@ -0,0 +1,872 @@
+бога, некоторые, конце, милый, the, рождения, трудно, добро, любви, больно, невозможно, спокойной, 
+слышишь, типа, получил, которое, приятель, хуже, никому, честь, успокойся, вашу, маленький, выглядит, 
+чарли, сына, неделю, i, девочка, делаю, шесть, ноги, история, рассказать, послушайте, часто, кстати, 
+двух, забудь, которых, следует, знают, пришла, семья, станет, матери, ребенок, план, проблем, например, 
+сделай, воды, немедленно, мира, сэм, телефон, перестань, правду, второй, прощения, ту, наше, уходи, твоих, 
+помоги, пол, внутри, нему, смог, десять, нашу, около, бывает, самого, большая, леди, сможем, вниз, легко, 
+делай, единственный, рада, меньше, волнуйся, хотим, полагаю, мам, иметь, своими, мере, наконец, начала, 
+минутку, работе, пожаловать, другого, двое, никакого, честно, школе, лучший, умереть, дам, насколько, 
+всей, малыш, оставить, безопасности, ненавижу, школу, осторожно, сынок, джо, таки, пытался, другое, б, 
+клянусь, машине, недели, стало, истории, пришлось, выглядишь, чему, сможет, купить, слышала, знали, 
+настоящий, сих, выйти, людям, замечательно, полиции, огонь, пойдём, спросить, дядя, детка, среди, особенно, 
+твоим, комнате, шоу, выпить, постоянно, делают, позвольте, родители, письмо, городе, случай, месяцев, мужик, 
+благодарю, o, ребенка, смешно, ответ, города, образом, любой, полностью, увидел, еду, имени, вместо, 
+абсолютно, обязательно, улице, твоё, убили, ваших, ехать, крови, решение, вина, поможет, своё, секунду, 
+обещаю, начать, голос, вещь, друзей, показать, нечего, э, месяц, подарок, приехал, самая, молодец, сделаем, 
+крайней, женщин, собираешься, конца, страшно, новости, идиот, потерял, спасти, вернуть, узнал, слушайте, 
+хотелось, сон, поняла, прошло, комнату, семь, погоди, главное, рано, корабль, пытаюсь, игра, умерла, 
+повезло, всему, возьму, таком, моем, глаз, настолько, идём, удачи, готова, семьи, садись, гарри, держись, 
+звучит, мило, война, человеком, право, такую, вопросы, представить, работаю, имеешь, красивая, идёт, никакой, 
+профессор, думает, войны, стала, стали, оттуда, известно, слышу, начал, подумать, позвонить, старый, придётся, 
+историю, вести, твоему, последнее, хочется, миллионов, нашла, способ, отношения, земле, фрэнк, получится, 
+говоря, связи, многие, пошёл, пистолет, убью, руках, получилось, президент, остановить, тьi, оставил, одним, 
+you, утра, боль, хорошие, пришёл, открой, брось, вставай, находится, поговорим, кино, людьми, полицию, покажу, 
+волосы, последние, брата, месяца
+"""
+
+
+fn gen_word_pairs[words: String = words_en]() -> List[String]:
+    var result = List[String]()
+    try:
+        var list = words.split(",")
+        for w in list:
+            var w1 = w[].strip()
+            for w in list:
+                var w2 = w[].strip()
+                result.append(w1 + " " + w2)
+    except:
+        pass
+    return result
+
+
+# ===----------------------------------------------------------------------===#
```

Comment:
**Question** Instead of copying the impl here, can we do something like

```
from builtin._hash.mojo import AHash
```

Reply:
Hm, I just rebased and wanted to try this out, but it seems like I can't run benchmarks at all. I get many errors like this one: 
```
******************** TEST 'Mojo Standard Library Benchmarks :: utils/bench_formatter.mojo' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 14: mojo /Users/mzaks/dev/mojo/stdlib/benchmarks/utils/bench_formatter.mojo -t
+ mojo /Users/mzaks/dev/mojo/stdlib/benchmarks/utils/bench_formatter.mojo -t
/Users/mzaks/dev/mojo/stdlib/benchmarks/utils/bench_formatter.mojo:17:6: error: unable to locate module 'benchmark'
from benchmark import Bench, BenchConfig, Bencher, BenchId, Unit, keep, run
     ^
mojo: error: failed to parse the provided Mojo source module
```

Reply:
Unfortunately, as a stdlib developer, that's a rough edge right now.  This is because, what you'd like to do, is have the `benchmark.mojopkg` get rebuilt (it depends on other parts of the stdlib that you may have modified, for example).  However, since its sources aren't available, it'll pick up the pre-built `benchmark.mojopkg` that ships with the Mojo nightly.

I've fixed up your benchmarks and removed the duplication of `AHash` and friends to just benchmark the old vs. new hash functions.  It should land today and we can iterate on future improvements (relocating the hash functions, defining the hasher trait, experimenting with replacing the existing hash function, changing the impl of `folded_multiply` to use the one from @soraros, etc.).